custom/icds_reports/static/js/directives/location-filter/location-filter.directive.js

```diff
@@ -132,7 +132,12 @@ function LocationFilterController($scope, $location, $uibModal, locationHierarch
 
     vm.userLocationId = userLocationId;
     vm.animationsEnabled = true;
-    vm.selectedLocationId = ['', 'undefined', 'null'].indexOf($location.search()['location_id']) === -1 ? $location.search()['location_id'] : vm.userLocationId;
+    var emptyValues = ['undefined', 'null'];
+    if (!haveAccessToAllLocations) {
```

Comment:
nit: it might be useful to have a comment here, or extract this into a named function. it's not intuitively obvious to me why this logic is here.